WasatchNET/EEPROM.cs

```diff
@@ -30,12 +30,12 @@ public class EEPROM : IEEPROM
 
         //internal const int MAX_PAGES = 256; // really 8, but last 2 are unallocated
         internal const int MAX_PAGES = 8;
-        const byte FORMAT = 9;
+        protected const byte FORMAT = 9;
 
-        Spectrometer spectrometer;
-        Logger logger = Logger.getInstance();
+        protected Spectrometer spectrometer;
+        protected Logger logger = Logger.getInstance();
 
-        public List<byte[]> pages { get; private set; }
+        public List<byte[]> pages { get; protected set; }
         public event EventHandler EEPROMChanged;

```

Comment:
This is not a change you need to do at this time.  However, down the road, I feel like "public event EventHandler" should be changed to "public event PropertyChangedEventHandler" (using System.ComponentModel).  We would then make EEPROM itself inherit from INotifyPropertyChanged.

Then in the individual Properties, blocks like this:
```
EventHandler handler = EEPROMChanged;
_field = value;
handler?.Invoke(this, new EventArgs());
```
can be replaced with this:
```
_field = value;
EEPROMChanged?.Invoke(this, new PropertyChangedEventArgs(nameof(field)));
```

This passes the name of the actual changed property to the recipient(s), so they can make their own decision as to whether they need to process the event or not based on whether it's a field they care about, and also how they wish to process it based on what was changed.  (As it is, they are notified of nothing other than "something in the EEPROM changed," requiring the entire EEPROM to be re-processed blindly).  It also includes a compile-time check that the passed property name is valid.  Also it's one line shorter :-)

Reply:
This is a very good note. This has always sort of half baked tbch...